### PR TITLE
lock-service: fix bug when closing rocksdb database

### PR DESCRIPTION
**Pull request by bmwill**
Frequently when running tests we'll see test failures with the following
error:

    libc++abi: Pure virtual function called!

This is due to the fact that the thread part of the lock-service runs on
may outlive the main thread leading to a race on closing the rocksdb
database vs cleaning up the C++ runtime, causing this segfault to occur
and preventing the database from being able to cleanly close.

This is fixed by joining the thread handle for the spawned lock-service
threads in the Drop impl of LockService.

### CODE REVIEW

**Comment by bmwill** [comment]
Alternatively joining on the spawned threads also fixes the issue. Although after looking more closely into what the lock service is i'm sort of skeptical as to why having this be a in a few separate threads vs just calling the locking methods themselves. The DB is threadsafe so i'm not sure I understand the benefit having this split out other than the increase in complexity.

**Review by gdanezis — APPROVED** [review]
Nice one!

File: crates/sui-storage/src/lock_service.rs [hunk 1]
```diff
@@ -271,8 +273,45 @@ const LOCKSERVICE_QUEUE_LEN: usize = 500;
...
+            .as_ref()
+            .expect("LockServiceInner should not have been dropped yet")
+    }
+}
+
+impl Drop for LockServiceInner {
+    fn drop(&mut self) {
+        self.sender.take();
+        self.query_sender.take();
+        self.run_command_loop
```
[the remark this answers is not in the document]
Comment (gdanezis) [reply crates/sui-storage/src/lock_service.rs:304]: Ah wait a second are these tokio, or OS tasks / threads? They are OS threads. So scratch that there is no abort.

**Review by velvia — APPROVED** [review]
@bmwill thanks, I'm fine with these changes.  Would you mind also combining with a few lines from my changes in my PR that 
1) use the telemetry-subscribers logger init line
2) Add a bit of logging?

File: crates/sui-storage/src/lock_service.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmwill) [reply crates/sui-storage/src/lock_service.rs:306]: Yeah these are OS threads

**Pull request merged**